### PR TITLE
Attempting to fix auth redirect/user store stuff

### PR DESCRIPTION
**Pull request by ckcollab**
# A brief description of the purpose of the changes contained in this PR.

Previously we'd redirect from dashboard to login, which would redirect to dashboard, which would redirect back.. causing a fail!

Trying to fix that with some better auth middleware guards and user store stuff, not doing great so far..!

# Hand testing
- [ ] add checklist here

# Checklist
- [ ] Code review by me
- [ ] New code covered by automated tests
- [x] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: src/frontend/middleware/auth.ts [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+    const user = userStore()
 
-    if (!csrftoken.value || auth.isLoggedIn?.value === false) {
-        csrftoken.value = null
+    // here I need to wait for the user to be fetched or not, which is "user.fetchUserHasBeenRan"
+    if (!user.fetchUserHasBeenRan) {
+
+      try {
+        // Wait for fetch to complete
+        await user.$state.fetchUserHasBeenRan
```
Comment (sw00d) [inline src/frontend/middleware/auth.ts]: is something fancy happening here? Why not just do `user.fetchUseHasBeenRan`? Does the `await` and `$state` actually listen to that property change or something?

File: src/frontend/package.json [hunk 2]
```diff
@@ -15,7 +15,6 @@
     },
     "devDependencies": {
         "@iconify/vue": "^4.1.1",
-        "@nuxt/devtools": "^1.0.0",
```
Comment (sw00d) [inline src/frontend/package.json:18]: ?? Dont like it anymore?

File: src/frontend/store/user.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            console.log("????????? error", error.value)
+            console.log("????????? data", data.value)
+
+            // const resp = await useRequest('/users/me/')
+            // const {execute} = useRequest('/users/me/')
+            // const resp = await execute()
+            // console.log("????????? resp", resp)
+            // const {data, error} = resp
+            // console.log("????????? error", error.value)
+            // console.log("????????? data", data.value)
```
Comment (sw00d) [inline src/frontend/store/user.ts]: pull?
```suggestion
```

File: src/frontend/middleware/auth.ts (hunk 1 above)
Comment (ckcollab) [reply src/frontend/middleware/auth.ts]: I think this was just an experiment and doesn't actually do anything?

File: src/frontend/package.json (hunk 2 above)
Comment (ckcollab) [reply src/frontend/package.json:18]: devtools integrated with nuxt now, not external dep, I believe!

https://github.com/nuxt/nuxt/releases/tag/v3.8.0

File: src/frontend/store/user.ts (hunk 3 above)
Comment (ckcollab) [reply src/frontend/store/user.ts]: def will, sorry, semi-draft PR :)

**Review by sw00d — APPROVED** [review]
Sexy

**Pull request merged**